contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,628 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title Aerodrome AMO strategy
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+
+import { ISugarHelper } from "../../interfaces/aerodrome/ISugarHelper.sol";
+import { INonfungiblePositionManager } from "../../interfaces/aerodrome/INonfungiblePositionManager.sol";
+import { IQuoterV2 } from "../../interfaces/aerodrome/IQuoterV2.sol";
+import { ISwapRouter } from "../../interfaces/aerodrome/ISwapRouter.sol";
+import { ICLPool } from "../../interfaces/aerodrome/ICLPool.sol";
+import { ICLGauge } from "../../interfaces/aerodrome/ICLGauge.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+
+import "hardhat/console.sol";
+
+contract AerodromeAMOStrategy is InitializableAbstractStrategy {
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+
+    /***************************************
+            Storage slot members
+    ****************************************/
+
+    /// @notice tokenId of the liquidity position
+    uint256 public tokenId;
+    /// @dev Cumulative amount of WETH + OETHb tokens present in Aerodrome Slipstream pool
+    ///      increase liquidity increases this number by the amount of tokens added
+    ///      decrease liquidity decreases this number by the amount of tokens removed
+    uint256 public netValue;
+    /// @notice the gauge for the corresponding Slipstream pool (clPool)
+    /// @dev can become an immutable once the gauge is created on the base mainnet
+    ICLGauge public clGauge;
+    /**
+     * @notice Specifies WETH to OETHb ratio the strategy contract aims for after rebalancing 
+     * in basis point format.
+     * 
+     * e.g. 2000 means 20% WETH 80% OETHb
+     */
+    uint256 public poolWethShare;
```

Comment:
If we are doing price based targeting, this should probably change to a target price rather than a target share.